### PR TITLE
Add krun support to podman machine

### PR DESCRIPTION
**Pull request by baude**
This PR adds libkrun support to podman machine.  This is an experimental feature and should not be marketed yet.  Before we unmark the experimental status on this function, we will need to have full CI support and a full podman point release has passed.

This work relies on the fact that vfkit and libkrun share a reasonably (if not perfectly) same API.  The --log-level debug option will not show a GUI screen for boots as krun is not capable of this.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by ashley-cui** [comment]
Tested init, start, ssh, run nginx,stop manually locally and they all seem to work. 

Read-over of code  LGTM too.

**Comment by rhatdan** [comment]
@jakecorrenti PTAL

**Review by jakecorrenti — COMMENTED** [review]
Some minor changes and nitpicks. Otherwise, LGTM

**Comment by rhatdan** [comment]
@slp PTAL

File: pkg/machine/vmconfigs/config_common.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ type QEMUConfig struct {
 // Stubs
 type AppleHVConfig struct{}
 type HyperVConfig struct{}
+type LibKRunConfig struct{}
```
Comment (Luap99) [inline pkg/machine/vmconfigs/config_common.go:22]: Should this be named `LibKrunConfig`? Why is the R uppercase?

File: pkg/machine/apple/apple.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+		// Here we are looping the mounts and for each mount, we are adding two unit files
+		// for virtiofs.  One unit file is the mount itself and the second is to automount it
+		// on boot.
+		autoMountUnit := parser.NewUnitFile()
+		autoMountUnit.Add("Automount", "Where", "%s")
+		autoMountUnit.Add("Install", "WantedBy", "multi-user.target")
+		autoMountUnit.Add("Unit", "Description", "Mount virtiofs volume %s")
+		autoMountUnitFile, err := autoMountUnit.ToString()
+		if err != nil {
+			logrus.Warnf(err.Error())
```
Comment (Luap99) [inline pkg/machine/apple/apple.go:83]: pre-existing but why the hell do we warn the err? If we cannot create a unit file this must error out properly. 
same in the others places below, if you do not want to this this in this PR it is fine for me as well but then lets create a issue for it

**Comment by slp** [comment]
Tested on M1 and M3 Pro, everything works fine, including GPU acceleration. I'd like to add proper support for `--log-level=debug` but I think it's better to do it in a different PR and get this one merged first. As for the code itself, I don't have anything to add on top of what @jakecorrenti and @Luap99 already pointed out.

File: pkg/machine/apple/apple.go (hunk 2 above)
Comment (baude) [reply pkg/machine/apple/apple.go:83]: ill add an error to the signature and tidy up!

Comment (Luap99) [reply pkg/machine/apple/apple.go:83]: you marked this as resolved but I still warnings here and below

Comment (baude) [reply pkg/machine/apple/apple.go:83]: well you did give me the option to NOT do it :)

Comment (baude) [reply pkg/machine/apple/apple.go]: fixed the other two, please merge

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**